pkg/cmd/pulumi/errors.go

```diff
@@ -89,11 +89,22 @@ func printDecryptError(e engine.DecryptError) {
 // Pulumi engine. This function is a type-specific handler for these errors that
 // prints out a panic-like banner with information about the error and how to
 // report it.
-func printSnapshotIntegrityError(err deploy.SnapshotIntegrityError) {
+func printSnapshotIntegrityError(err error, sie deploy.SnapshotIntegrityError) {
+	readOrWrite := ""
+	if sie.Op == deploy.SnapshotIntegrityRead {
+		readOrWrite = `
+NOTE: This error occurred while reading a snapshot. This means that it is likely
+that a *previous* operation which *wrote* the snapshot was the cause of the issue.
+If you are able to find the operation that wrote the snapshot, please report details
+of *that* run instead. If you are unable to determine the cause, please report this
+error as-is.
```

Comment:
> a *previous* operation which *wrote* the snapshot was the cause of the issue

and

> If you are unable to determine the cause, please report this
> error as-is.

I could see users interpreting this as us requesting that they find the underlying root cause of the error, and users aren't going to be able to do that and will likely continue reporting this error as-is.

Wondering if we should tweak the message a bit to make it more clear that a previous run introduced the snapshot integrity issue and having details of that run will help us track down the underlying problem.

Reply:
@justinvp thoughts on the updated copy?

```
The Pulumi CLI encountered a snapshot integrity error. This is a bug!

================================================================================
We would appreciate a report: https://github.com/pulumi/pulumi/issues/

NOTE: This error occurred while reading a snaphot. This error was introduced by
a previous operation when it wrote the snapshot. If you have details of this
operation, please include them in your report as well.

Please provide all of the below text in your report.
================================================================================
```